### ISSUE TITLE
Upgrade Hazelcast to 5.5.0

### ISSUE THREAD

**Issue by guusdk** [igniterealtime/openfire-hazelcast-plugin#101]
Hazelcast 5.5.0 has been released. This plugin should be upgraded to to this version.

5.5.0 brings various improvement, but is also the first version to be designated 'LTS': https://docs.hazelcast.com/hazelcast/5.5/migrate/lts

**Comment by guusdk** [comment]
Latest release to date seems to be 5.5.2

The release notes between 5.5.0 and 5.5.2 show mostly bugfixes and new features, without listing known issues.

**Comment by guusdk** [comment]
In an ... interesting ... move, Hazelcast is no longer publishing patch releases (aka 'bugfixes') for their community edition: https://hazelcast.com/blog/changes-to-community-edition/

Seems that we're stuck on 5.5.0.

As an aside: this doesn't bode well for the future, if you're asking me. We may want to spend some time at looking at alternatives to Hazelcast.

**Comment by GregDThomas** [comment]
> this doesn't bode well for the future, if you're asking me. We may want to spend some time at looking at alternatives to Hazelcast.

Agreed. Companies seem to be scaling back more and more their commitment to open source. 

Off the top of my head, I wonder if we (*) actually need a clustering technology. There's a shared data source already (the database) which could be used to share data, and something like gRPC (or XMPP!) could be used to send messages between cluster members to trigger events such as 'cluster member joined', 'user joined chat', etc. etc.

I don't know if all the work identified in https://github.com/igniterealtime/openfire-hazelcast-plugin/issues/50 was ever implemented - if not completing that would probably be a first step anyway, that would make it clearer to see what's actually left to replace.

(*) I saw we, my input nowadays is pretty much limited to being picky on log messages.
